### PR TITLE
Add DuckDBConfig to handle the system resource configurations

### PR DESCRIPTION
**Pull request by goldmedal**
# Description
There're 3 configurations exposed by this PR:
- `duckdb.memory-limit`: To set `memory_limit` in DuckDB. The default value is `the runtime max heap size / 2`
- `duckdb.home-directory`: To set `home-directory` in DuckDB.
  - This config is disable because the known issue: https://github.com/duckdb/duckdb/issues/10062
-  `duckdb.temp-directory`: To set `temp-driectory` in DuckDB.

### CODE REVIEW

**Review by AShiou — APPROVED** [review]

File: accio-base/src/main/java/io/accio/base/client/duckdb/DuckDBConfig.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ */
+
+package io.accio.base.client.duckdb;
+
+import io.airlift.configuration.Config;
+
+public class DuckDBConfig
+{
+    private String memoryLimit = Runtime.getRuntime().maxMemory() / 2 + "B";
+    private String homeDirectory = "/tmp/duckdb";
```
[the remark this answers is not in the document]
Comment (goldmedal) [reply accio-base/src/main/java/io/accio/base/client/duckdb/DuckDBConfig.java:22]: hmm.. or we don't give the default value for it? It's hard to decide where to place it.

File: accio-main/src/main/java/io/accio/main/web/CacheResource.java [hunk 2]
```diff
@@ -99,7 +99,7 @@ public void getDuckDBSettings(@Suspended AsyncResponse asyncResponse)
             asyncResponse.resume(Response.ok(cacheManager.getDuckDBSettings()).build());
         }
         catch (Exception e) {
-            asyncResponse.resume(Response.serverError().build());
+            asyncResponse.resume(asyncResponse.resume(e));
```
Comment (brandboat) [inline accio-main/src/main/java/io/accio/main/web/CacheResource.java:102]: one last comment, perhaps we can omit the try-catch block in this section. It seems that the AccioExceptionMapper is set up to catch the error and return a 500 response in its toResponse method. However, I'm not entirely certain about this, could you help me check this?

Comment (goldmedal) [reply accio-main/src/main/java/io/accio/main/web/CacheResource.java:102]: Yes, If we throw the runtime exception from `cacheManager.getDuckDBSettings`, we can omit the try-catch.

**Review by brandboat — APPROVED** [review]
LGTM

**Comment by grieve54706** [comment]
@goldmedal Do we check the path if it does not exist and the feature still works? I tried the home_directory with a non-exist will not work. However, using an existing path will work in the session.

I also check your PR in the DuckDB about `home_directory`. I tried the way `SET home_directory; SELECT current_setting('home_directory');` in one statement and it got the value. But I tried another way `SET home_directory;` and `SELECT current_setting('home_directory');` with different connections will not work.

**Comment by goldmedal** [comment]
> @goldmedal Do we check the path if it does not exist and the feature still works? I tried the home_directory with a non-exist will not work. However, using an existing path will work in the session.
> 
I think we don't need to check it. We just pass the config to DuckDB. This config should checked by DuckDB. 

> I also check your PR in the DuckDB about `home_directory`. I tried the way `SET home_directory; SELECT current_setting('home_directory');` in one statement and it got the value. But I tried another way `SET home_directory;` and `SELECT current_setting('home_directory');` with different connections will not work.

I has checked this issue is a bug or behavior of DuckDB. https://github.com/duckdb/duckdb/issues/10062
That's why I disable the feature temporarlly.
https://github.com/Canner/accio/blob/33442e032219eb35da44ae1454a6c614a5d89768/accio-base/src/main/java/io/accio/base/client/duckdb/DuckdbClient.java#L75-L77

**Pull request merged**